.github/workflows/nightly-masternet-deploy.yml

```
@@ -9,13 +9,28 @@ concurrency:
   cancel-in-progress: false
 
 jobs:
+  get-latest-commit:
+    runs-on: ubuntu-latest
+    outputs:
+      commit: ${{ steps.get_commit.outputs.COMMIT }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get latest published commit
+        id: get_commit
+        run: |
+          COMMIT=$(curl -s https://registry.hub.docker.com/v2/repositories/aztecprotocol/aztec/tags/master | jq ".digest" | tr -d '"')
```

Optional suggestion to use raw mode to remove double quotes but works either way.

```suggestion
          COMMIT=$(curl -s https://registry.hub.docker.com/v2/repositories/aztecprotocol/aztec/tags/master | jq -r ".digest")
```